### ISSUE TITLE
modules_to_save有bug

### ISSUE THREAD

**Issue by Sakura-not-sleep** [modelscope/ms-swift#3773]
我在lora微调的时候，又在llm里面新加入了一个线性层，名字是self.classifier，在sh文件里面我设置了  --modules_to_save classifier \
但是在训练过程中，出现了梯度更新异常，

![Image](https://github.com/user-attachments/assets/fe85837c-5c93-4071-8460-d5e9279bc374)
奇怪的是，参数已经更新了，但是下面的打印梯度却是nan，紧接着，在这个下面参数却仍然是没有更新的样子，而且还出现更新后的模块都是nan，

![Image](https://github.com/user-attachments/assets/f704a0c3-0237-4fb4-a999-f9097231fdea)

这是什么情况呢，是我对自定义模块的初始化不对吗，我应该如何对自定义的几个linear进行初始化呢。

**Comment by Sakura-not-sleep** [comment]
这个是我的新模块的参数初始化，是不是哪里有问题

![Image](https://github.com/user-attachments/assets/67ea2230-db80-4d0e-983f-68a1fa3c9290)

**Comment by Sakura-not-sleep** [comment]
这个是训练脚本CUDA_VISIBLE_DEVICES=4,5 \
VIDEO_MAX_PIXELS=50176 \
FPS_MAX_FRAMES=12 \
swift sft \
    --model /data/ji/InternVL2.5/OpenGVLab/InternVL2_5-8B \
    --dataset /data/ji/InternVL2.5/mydata/train.jsonl \
    --val_dataset /data/ji/InternVL2.5/mydata/test.jsonl \
    --acc_strategy seq \
    --train_type lora \
    --torch_dtype bfloat16 \
    --attn_impl flash_attn \
    --num_train_epochs 5 \
    --per_device_train_batch_size 2 \
    --split_dataset_ratio 0.0 \
    --per_device_eval_batch_size 1 \
    --learning_rate 8e-5 \
    --lora_rank 8 \
    --lora_alpha 32 \
    --target_modules all-linear \
    --freeze_vit true \
    --freeze_aligner true \
    --gradient_accumulation_steps 4 \
    --save_strategy epoch \
    --eval_strategy epoch \
    --eval_steps 1 \
    --save_steps 1 \
    --save_total_limit 5 \
    --logging_steps 10 \
    --max_length 2048 \
    --modules_to_save classifier \
    --output_dir /data1_hdd/ji/InternVL2.5/_8B \
    --warmup_ratio 0.1 \
    --seed 2025 \
    --temperature 0 \
    --dataloader_num_workers 4

**Comment by Lsjay3** [comment]
同问，请问有结果了吗？

还想问下我想解冻["score", "embed_tokens"]加入训练，在sh中该如何传参呢？

**Comment by cnyvfang** [comment]
hi, 我也有类似的问题，请问解决了吗

**Comment by Lsjay3** [comment]
来信收悉，顺祝恭安！

**Comment by Jintao-Huang** [comment]
请检查 新加入层参数的初始化情况
